src/sponsorableContributorsFetcher.ts

```diff
@@ -0,0 +1,114 @@
+import { Config } from './config';
+import { ContributionCountOfUserIntoRepo, RepositoryName } from './contributorFetcher';
+import { graphql } from '@octokit/graphql';
+import { GraphQlQueryResponseData, RequestParameters } from '@octokit/graphql/dist-types/types';
+
+export interface Sponsor {
+    __typename: string;
+    email?: string;
+    login: string;
+    url: string;
+    sponsorsListing: {
+        name: string | null;
+        dashboard: string | null;
+    };
+}
+
+// query WIP: fetchiing first 10 sponsors, need to add pagination
+export const queryTemplate = `query fetchSponsorable($queryString: String!) {
+  search(query: $queryString, type: USER, first: 10) {
+    nodes {
+      ... on User {
+        __typename
+        email
+        login
+        url
+        sponsorsListing {
+          name
+          dashboardUrl
+        }
+      }
+        ... on Organization {
+        __typename
+        login
+        url
+        sponsorsListing {
+          name
+          dashboardUrl
+        }
+      }
+    }
+  }
+}`;
+
+interface Variables extends RequestParameters {
+    queryString: string;
+}
+
+export class SponsorableContributorsFetcher {
+    private readonly config: Config;
+
+    public constructor(config: Config) {
+        this.config = config;
+    }
+
+    public async fetchSponsorableContributorsInformation(
+        token: string,
+        query: string,
+        contributors: Map<RepositoryName, ContributionCountOfUserIntoRepo[]>
+    ): Promise<Map<string, Sponsor[]>> {
+        const allSponsorableUsersInfo = [];
+        const sponsorables = new Map<string, Sponsor[]>();
+
+        for (const [repoIdentifier, githubUsers] of contributors.entries()) {
+            console.log(`for repo: ${repoIdentifier}`);
+            for (const contributor of githubUsers) {
+                const userLogin = contributor.login;
+
+                const variables: Variables = { queryString: userLogin };
+                const response = await this.fetchSponsorableContributor(token, variables, query);
+
+                for (const user of response) {
+                    if (user.sponsorsListing?.name && user.__typename === 'User') {
+                        allSponsorableUsersInfo.push({ repoId: contributor.repoIdentifier, user });
+                    }
+                }
+            }
+
+            // need to refactor:
```

Comment:
I suspect this would be simpler if you built the map inside the loop above, instead of building it there an array, and then converting it to a map here.

Reply:
I was having issues trying to do that inside the loop, (was getting duplicates)